stl/inc/xlocnum

```diff
@@ -750,366 +748,363 @@ private:
-            }
 
-            if (_Phex < 0) { // put one back
-                *_Ptr++ = '0';
-                ++_Phex;
-            }
-        }
-
-        for (; _First != _Last && (_Idx = _Find_elem(_Atoms, *_First)) < _Numget_signoff;
-             _Seendigit = true, (void) ++_First) {
-            if (_Significant < _Max_sig_dig) { // save a significant fraction digit
-                *_Ptr++ = _Src[_Idx];
-                ++_Significant;
-            }
-        }
-
-        if (_Seendigit && _First != _Last
-            && (*_First == _Atoms[_Numget_poff + 1]
-                || *_First == _Atoms[_Numget_poff])) { // 'p' or 'P', collect exponent
-            *_Ptr++ = 'p';
-            ++_First;
-            _Seendigit   = false;
-            _Significant = 0;
-
-            if (_First != _Last) {
-                if (*_First == _Atoms[_Numget_signoff + 1]) { // gather plus sign
-                    *_Ptr++ = '+';
-                    ++_First;
-                } else if (*_First == _Atoms[_Numget_signoff]) { // gather minus sign
-                    *_Ptr++ = '-';
-                    ++_First;
+                if (_Seendigit) {
+                    *_Ptr++ = '0'; // put one back
                 }
-            }
 
-            for (; _First != _Last && *_First == _Atoms[0]; ++_First) { // strip leading zeros
-                _Seendigit = true;
-            }
-
-            if (_Seendigit) {
-                *_Ptr++ = '0'; // put one back
+                for (; _First != _Last && (_Idx = _STD _Find_elem(_Atoms, *_First)) < _Dec_digits_cnt;
+                     _Seendigit = true, (void) ++_First) {
+                    if (_Significant < _MAX_EXP_DIG) { // save a significant exponent digit
+                        *_Ptr++ = _Src[_Idx];
+                        ++_Significant;
+                    }
+                }
             }
 
-            for (; _First != _Last && (_Idx = _Find_elem(_Atoms, *_First)) < _Numget_signoff;
-                 _Seendigit = true, (void) ++_First) {
-                if (_Significant < _MAX_EXP_DIG) { // save a significant exponent digit
-                    *_Ptr++ = _Src[_Idx];
-                    ++_Significant;
-                }
+            if (_Bad || !_Seendigit) {
+                _Ptr = _Ac; // roll back pointer to indicate failure
             }
-        }
 
-        if (_Bad || !_Seendigit) {
-            _Ptr = _Ac; // roll back pointer to indicate failure
+            *_Ptr = '\0';
+            return _Pten;
         }
+    }
 
-        *_Ptr     = '\0';
-        *_Phexexp = _Phex; // power of 16 multiplier
-        return 0; // power of 10 multiplier
+    int __CLRCALL_OR_CDECL _Getffldx(char* _Ac, _InIt& _First, _InIt& _Last, ios_base& _Iosbase, int* _Phexexp) const {
+        // TRANSITION, ABI, unused now
+        // get hex floating-point field from [_First, _Last) into _Ac
+        return _Getffld(_Ac, _First, _Last, _Iosbase, _Phexexp);

```

Comment:
Now I'm leaving `_Getffld` unused and making `do_get` overloads call `_Getffldx` intead.

Reply:
Reactivating this comment as we talked about this in the weekly maintainer meeting and agree that mix-and-match ABI is a significant concern here, so we need to double-check that everything is okay. If changes are necessary, it should be possible to add a new function (either a free function or a templated member function to avoid affecting the export surface) that existing functions call, instead of having existing functions call each other - then there would be no mix-and-match issues.

The last time we dealt with this scenario, I added the `_ENABLE_V2_BEHAVIOR` logic, but that was extremely unusual/complicated (there are literally no other occurrences of that technique in the STL) and I would prefer to avoid doing that again.

Reply:
> If changes are necessary, it should be possible to add a new function (either a free function or a templated member function to avoid affecting the export surface) that existing functions call, instead of having existing functions call each other - then there would be no mix-and-match issues.

😂Oh, this is what I chose to do later. I'm sorry for not mentioning this here.